README.md

```diff
@@ -95,6 +95,17 @@ See memory requirements for the latest release [here](https://docs.aws.amazon.co
 
 The [corePKCS11](https://github.com/FreeRTOS/corePKCS11) library is an implementation of the PKCS #11 interface (API) that makes it easier to develop applications that rely on cryptographic operations. Only a subset of the [PKCS #11 v2.4](https://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html) standard has been implemented, with a focus on operations involving asymmetric keys, random number generation, and hashing.
 
+The Cryptoki or PKCS #11 standard defines a platform-independent API to manage and use cryptographic tokens. The name, "PKCS #11", is used interchangeably to refer to the API itself and the standard which defines it.
+
+The PKCS #11 API is useful for writing software without taking a dependency on any particular implementation or hardware. By writing against the PKCS #11 standard interface, code can be used interchangeably with multiple algorithms, implementations and hardware. 
+
+The purpose of corePKCS11 is to allow rapid prototyping and development. A [porting guide](https://docs.aws.amazon.com/freertos/latest/portingguide/afr-porting-pkcs.html) is available for implementing corePKCS11 on new platforms. 
+The goal of this library is to allow for exploring and building on the PKCS #11 itnerface, with out having to commit to any specific hardware. 
+Once a secure cryptoprocessor such as a TPM, HSM, Secure Element or other hardware is chosen, the PKCS #11 implementation associated with that hardware should instead be used. If this library is used alongside a cryptoprocessor, the benefits of said cryptoprocessor will be lost.
+
+Since the PKCS #11 interface is defined as part of the PKCS #11 [specification](https://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html) replacing this library with another implementation 
```

Comment:
`this`->`corePKCS11` makes the reference clear
```suggestion
Since the PKCS #11 interface is defined as part of the PKCS #11 [specification](https://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html) replacing the corePKCS11 library with another implementation 
```